File: 04-factorial/Factorial.cs
// C# program to compute factorial of a given number
// Written by M.V.Harish Kumar on 06-01-2024
using System;

namespace Factorial
{
    class Factorial
    {
        static ulong Fact(ulong n)
        {
            return n <= 1  ? 1 : n * Fact(n-1);
        }

        static void Main(string[] args)
        {
            Console.Write("Enter value for `n`: ");
            ulong n = Convert.ToUInt64(Console.ReadLine());
            Console.WriteLine($"{n}! = {Fact(n)}");
        }
    }
}

File: 04-factorial/factorial.ml
(* Ocaml program to compute factorial *)
(* Written by M.V.Harish Kumar on 28-05-2024 *)
open Printf
let fact n =
  let rec fact_tr n a =
    if n < 1 then a else fact_tr (n-1) (n*a)
  in fact_tr n 1

let () =
  print_string "Enter value for `n`: ";
  let n = read_int () in
    printf "%d! = %d\n" n (fact n)